src/pages/Compare/index.spec.tsx

```diff
@@ -127,9 +152,192 @@ describe(__filename, () => {
+    });
+
+    expect(dispatch).not.toHaveBeenCalled();
+  });
+
+  it('dispatches fetchVersion() on update if base version is different', () => {
+    const addonId = 123456;
+    const baseVersion = fakeVersion;
+    const headVersion = { ...fakeVersion, id: baseVersion.id + 1 };
+
+    const store = configureStore();
+    const fakeThunk = createFakeThunk();
+    const _fetchVersion = fakeThunk.createThunk;
+    const dispatch = spyOn(store, 'dispatch');
+
+    const root = render({
+      _fetchVersion,
+      ...getRouteParams({
+        addonId,
+        baseVersionId: baseVersion.id - 10,
+        headVersionId: headVersion.id,
+      }),
+      store,
+    });
+
+    dispatch.mockClear();
+    _fetchVersion.mockClear();
+    root.setProps({
+      match: {
+        params: getRouteParams({
+          addonId,
+          baseVersionId: baseVersion.id,
+          headVersionId: headVersion.id,
+        }),
+      },
+    });
+
+    expect(dispatch).toHaveBeenCalledWith(fakeThunk.thunk);
+    expect(_fetchVersion).toHaveBeenCalledWith({
+      addonId,
+      versionId: baseVersion.id,
+    });
+  });
+
+  it('dispatches fetchVersion() on update if head version is different', () => {
+    const addonId = 123456;
+    const baseVersion = fakeVersion;
+    const headVersion = { ...fakeVersion, id: baseVersion.id + 1 };
+
+    const store = configureStore();
+    const fakeThunk = createFakeThunk();
+    const _fetchVersion = fakeThunk.createThunk;
+    const dispatch = spyOn(store, 'dispatch');
+
+    const root = render({
+      _fetchVersion,
+      ...getRouteParams({
+        addonId,
+        baseVersionId: baseVersion.id,
+        headVersionId: headVersion.id + 10,
+      }),
+      store,
+    });
+
+    dispatch.mockClear();
+    _fetchVersion.mockClear();
+    root.setProps({
+      match: {
+        params: getRouteParams({
+          addonId,
+          baseVersionId: baseVersion.id,
+          headVersionId: headVersion.id,
+        }),
+      },
+    });
+
+    expect(dispatch).toHaveBeenCalledWith(fakeThunk.thunk);
+    expect(_fetchVersion).toHaveBeenCalledWith({
+      addonId,
+      versionId: baseVersion.id,
+    });
+  });
+
+  it('dispatches fetchVersion() on update if addon ID is different', () => {
+    const addonId = 123456;
+    const baseVersion = fakeVersion;
+    const headVersion = { ...fakeVersion, id: baseVersion.id + 1 };
+
+    const store = configureStore();
+    const fakeThunk = createFakeThunk();
+    const _fetchVersion = fakeThunk.createThunk;
+    const dispatch = spyOn(store, 'dispatch');
+
+    const root = render({
+      _fetchVersion,
+      ...getRouteParams({
+        addonId: addonId + 10,
+        baseVersionId: baseVersion.id,
+        headVersionId: headVersion.id,
+      }),
+      store,
+    });
+
+    dispatch.mockClear();
+    _fetchVersion.mockClear();
+    root.setProps({
+      match: {
+        params: getRouteParams({
+          addonId,
+          baseVersionId: baseVersion.id,
```

Comment:
Neither `baseVersionId` or `headVersionId` seem necessary for this test so if you can move them both to defaults in the helpers then that would make the test easier to read.

Reply:
it is actually tricky to remove because the values of `addonId` and `baseVersionId` are used in the `expect()` call right after. `headVersionId` could be omitted though.